source/app-development/tutorials-interactive-apps/k8s-like-hpc-jupyter.rst

```diff
@@ -0,0 +1,371 @@
+so it skips a lot of important details about app development that may be found in
+other tutorials like :ref:`app-development-tutorials-interactive-apps-add-jupyter`.
+
+
+We're going to be looking at the `bc osc jupyter`_ app which you can fork, clone
+and modify for your site.  This page also holds the `submit yml in full`_ for reference.
+
+The container
+-------------
+
+The container to make Kubernetes pods look like HPC compute would end up needing all
+the OS packages present on the HPC environment.  The OS of the container itself would also
+need to compatible with packages installed on HPC environment so if you run RHEL 8 on HPC
+you would also need to run RHEL 8 inside the container.
+
+Things like Lmod and HPC applications will need to be run inside the Pod's container just like
+if a job was spawned in the traditional HPC resource manager.
+
+
+Switch between SLURM and Kubernetes
+-----------------------------------
+
+The first big change from a traditional HPC interactive app is the main YAML structure is wrapped
+in a large ``if`` statement based on the cluster choice. If a user choose one of the HPC clusters,
+the SLURM submit YAML is rendered, otherwise the Kubernetes YAML is rendered.
+
+Here is the beginning of the block:
+
+.. code-block:: text
+
+  <% if cluster =~ /owens|pitzer/ -%>
+  ---
+  batch_connect:
+    template: "basic"
+    conn_params:
+      - jupyter_api
+  script:
+  ...SLURM specific...
+
+
+Here is the logic to select Kubernetes:
+
+.. code-block:: text
+
+  <% elsif cluster =~ /kubernetes/
+  ...Ruby variables setup here...
+
+container spec
+--------------
+
+Let's look at this section first of the Kubernetes block.  Here you must specify the ``name``, ``image``
+and ``command``.  The name determines the Pod Id (the job name in HPC parlance).
+
+The ``image`` should be the HPC container image and ``command`` will be the job script that has been adapted to
+work with both SLURM and Kubernetes.  The ``command`` will be run from the user's home directory and will cover mount
+requirements in :ref:`mount requirements <kubernetes-mount-requirements>`.
+
+Next you can specify additional environment variables in ``env``. 
+
+.. code-block:: yaml
+
+  native:
+    container:
+      name: "jupyter"
+      image: "docker-registry.osc.edu/ondemand/ondemand-base-rhel7:0.3.1"
+      image_pull_policy: "IfNotPresent"
+      command: ["/bin/bash","-l","<%= staged_root %>/job_script_content.sh"]
+      restart_policy: 'OnFailure'
+      env:
+        NB_UID: "<%= Etc.getpwnam(ENV['USER']).uid %>"
+        NB_USER: "<%= ENV['USER'] %>"
+        NB_GID: "<%= Etc.getpwnam(ENV['USER']).gid %>"
+        CLUSTER: "<%= compute_cluster %>"
+
+
+Here is the default environment. You can use a ``null`` here to unset any of these.
+
+.. code-block:: text
+
+  USER: username,
+  UID: run_as_user,
+  HOME: home_dir,
+  GROUP: group,
+  GID: run_as_group,
+  KUBECONFIG: '/dev/null'
+
+
+resource requests
+-----------------
```

Comment:
I don't think we need this bit about resources either.

Reply:
Removed